common/get_prices.go

```diff
@@ -0,0 +1,130 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// MainnetEndpoint is a mainnet Ethereum endpoint.
+// from https://chainlist.org/chain/1
+const MainnetEndpoint = "https://eth-rpc.gateway.pokt.network"
+
+var (
+	functionSig = ethcrypto.Keccak256([]byte("latestRoundData()"))[:4]
+
+	errUnsupportedNetwork = errors.New("unsupported network")
+)
+
+// see latestRoundData at https://docs.chain.link/data-feeds/price-feeds/api-reference/
+func latestRoundDataReturnArgs() abi.Arguments {
+	uint256Ty, err := abi.NewType("uint256", "", nil)
+	if err != nil {
+		panic(fmt.Errorf("failed to create uint256 type: %w", err))
+	}
+
+	int256Ty, err := abi.NewType("int256", "", nil)
+	if err != nil {
+		panic(fmt.Errorf("failed to create int256 type: %w", err))
+	}
+
+	uint80Ty, err := abi.NewType("uint80", "", nil)
+	if err != nil {
+		panic(fmt.Errorf("failed to create uint80 type: %w", err))
+	}
+
+	return abi.Arguments{
+		{
+			Type: uint80Ty,
+		},
+		{
+			Type: int256Ty,
+		},
+		{
+			Type: uint256Ty,
+		},
+		{
+			Type: uint256Ty,
+		},
+		{
+			Type: uint80Ty,
+		},
+	}
+}
+
+// GetETHUSDPrice returns the current ETH/USD price from the Chainlink oracle.
+// It errors if the chain ID is not the Ethereum mainnet.
+func GetETHUSDPrice(ctx context.Context, ec *ethclient.Client) (*big.Int, error) {
+	chainID, err := ec.ChainID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var to ethcommon.Address
+
+	switch chainID.Uint64() {
+	case 1:
+		// see https://data.chain.link/ethereum/mainnet/crypto-usd/eth-usd
+		to = ethcommon.HexToAddress("0x5f4ec3df9cbd43714fe2740f5e3616155c5b8419")
+	case 5, 1337, 31337:
```

Comment:
Maybe we should have constants so people know what these are? Is this Goerli, Ganache, and Hardhat? We'll need at least some of them for the startup check you created a ticket for.